tests/test_testsignalgenerator.py

```diff
@@ -120,3 +120,22 @@ def test_lumasteps_raises_on_invalid_format(self):
 
         with self.assertRaises(ValueError):
             next(UUT)
+
+    def test_lumasteps_with_step_2(self):
+        """Testing that the LumaSteps generator produces alternate frames when skip is set"""
+        width = 240
+        height = 4
+        UUT = LumaSteps(src_id, flow_id, width, height,
+                        origin_timestamp=origin_timestamp,
+                        cog_frame_format=CogFrameFormat.S16_422_10BIT,
+                        rate=Fraction(50, 1),
+                        step=2)
+
+        # Extracts the first 10 grains from the generator
+        grains = [grain for _, grain in zip(range(10), UUT)]
+
+        ts = origin_timestamp
+        for grain in grains:
+            self.assertEqual(grain.origin_timestamp, ts)
+            self.assertEqual(grain.sync_timestamp, ts)
+            ts = Timestamp.from_count(ts.to_count(25, 1) + 1, 25, 1)
```

Comment:
The 25/50 magic number is quite hard to follow here; pulling `rate = Fraction(50,1)` and `step = 2` out and then using `ts.to_count(rate) + step` would be clearer.